lib/ohai/plugins/linux/virtualization.rb

```diff
@@ -188,6 +216,18 @@ def nova_exists?
         # But, this could overwrite virtualization[:systems][:lxc] = "guest"
         # If so, we may need to look further for a differentiator (OHAI-573)
         virtualization[:systems][:lxc] = "host"
+      elsif init_environ_content =~ %r{\x00?container=lxc\x00?}
```

Comment:
I was just starting to implement this myself because I realized lxc detection stopped working when I upgraded to 16.04 hosts. Here was my solution:
```
init_environ = File.read('/proc/1/environ').split("\0".b).map { |x| x.split('=', 2) }.to_h
if init_environ['container']
  ...
```
This splits the data by the null bytes, and then splits each environment variable statement into keys and values in a hash. Might be overkill if LXC is the only thing setting useful info in the init environment though.